### ISSUE TITLE
Using Xamarin.Forms Previewer with Page that has ViewModelLocaltor enable throws at design-time

### ISSUE THREAD

**Issue by kdawg1406** [PrismLibrary/Prism#1037]
Using Xamarin.Forms Previewer with Page that has ViewModelLocaltor enable throws at design-time

Create a new XF project using Prism

Enable ViewModelLocator on the MainPage

Build the solution

Open the VS XF Previewer

Exception throw.

**Discussion:**
VML needs to handle design-time scenarios in XF.  The code can't throw an exception at design-time.  If Xamarin can't provide reliable design-time API, this code will have to handle this scenario and not throw.

**Comment by dansiegel** [comment]
@kdawg1406 I'm not sure what exactly you're doing but, I can File -> New a project add the ViewModelLocator to a View and I get no exception thrown by the Xamarin Forms Previewer. 

Can you provide some more information like which versions of Prism, Xamarin Forms and your IDE.

**Comment by kdawg1406** [comment]
Use AppMap to create the solution.

**Comment by dansiegel** [comment]
@kdawg1406 if you remove the ViewModelLocator does the Previewer in VS succeed?

**Comment by kdawg1406** [comment]
Yes

**Comment by dansiegel** [comment]
Ok... with the Attached property removed try the following.

```cs
public class ViewModelLocatorExtension : IMarkupExtension
{
    public object View { get; set; }

    public object ProvideValue(IServiceProvider serviceProvider)
    {
        object viewModel = null;
        ViewModelLocationProvider.AutoWireViewModelChanged(View, (v, vm) => viewModel = vm);
        return viewModel;
    }
}
```

```xml
xmlns:xaml="{local import to the above class}"
x:Name="view"
BindingContext="{xaml:ViewModelLocator {x:Reference view}}"
```

**Comment by brianlagunas** [comment]
The default behavior works fine for me.  The only issue that was noticed is that the App.xaml.cs requires an empty default ctor.  I added this, and everything worked fine:

`public App() : this(null) { }`

I found no issues of exceptions in the previewer with VML.  The bindings worked and displayed all the default values defined for the properties.

When PrismAplication was written, I did not consider anything creating it using reflection, so I was not concerned with having a default ctor.  Since it is obvious that the Previewer needs this, I will update the PrismApplication class to have a default ctor.
